internal/internal.go

```diff
@@ -224,13 +229,25 @@ func (p *envoyExtAuthzGrpcServer) Check(ctx ctx.Context, req *ext_authz.CheckReq
 
 	logrus.WithFields(logrus.Fields{
 		"query":               p.cfg.Query,
+		"dry-run":             p.cfg.DryRun,
 		"decision":            result.decision,
 		"err":                 err,
 		"txn":                 result.txnID,
 		"metrics":             result.metrics.All(),
 		"total_decision_time": time.Since(start),
 	}).Info("Returning policy decision.")
 
+	// If dry-run mode, override the Status code to unconditionally Allow the request
+	// DecisionLogging should reflect what "would" have happened
+	if p.cfg.DryRun {
```

Comment:
I think we should defer this block at the top of the function so that errors due not trigger denial when dry-run is enabled. /cc @ashutosh-narkar